Sources/Basics/GitURL.swift

```diff
@@ -0,0 +1,44 @@
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+public struct GitURL: Encodable, Equatable, Hashable, Sendable {
+    private let urlString: String
+
+    public init(stringLiteral: String) {
```

Comment:
probably worth adding syntatic sugar for `init(_ string: String)` to make call sites nicer